pkg/deploy/lattice/access_log_subscription_synthesizer.go

```diff
@@ -0,0 +1,62 @@
+package lattice
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/aws-application-networking-k8s/pkg/model/core"
+	model "github.com/aws/aws-application-networking-k8s/pkg/model/lattice"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils/gwlog"
+)
+
+type accessLogSubscriptionSynthesizer struct {
+	log                          gwlog.Logger
+	client                       client.Client
+	accessLogSubscriptionManager AccessLogSubscriptionManager
+	stack                        core.Stack
+}
+
+func NewAccessLogSubscriptionSynthesizer(
+	log gwlog.Logger,
+	client client.Client,
+	accessLogSubscriptionManager AccessLogSubscriptionManager,
+	stack core.Stack,
+) *accessLogSubscriptionSynthesizer {
+	return &accessLogSubscriptionSynthesizer{
+		log:                          log,
+		client:                       client,
+		accessLogSubscriptionManager: accessLogSubscriptionManager,
+		stack:                        stack,
+	}
+}
+
+func (s *accessLogSubscriptionSynthesizer) Synthesize(ctx context.Context) error {
+	var accessLogSubscriptions []*model.AccessLogSubscription
+	err := s.stack.ListResources(&accessLogSubscriptions)
+	if err != nil {
+		return err
+	}
+
+	var ret error = nil
+	for _, als := range accessLogSubscriptions {
+		if !als.Spec.IsDeleted {
+			s.log.Debugf("Started creating or updating access log subscription %s", als.ID())
+			_, err := s.accessLogSubscriptionManager.Create(ctx, als)
+			if err != nil {
+				s.log.Debugf("Synthesizing access log subscription %s failed due to %s", als.ID(), err)
+				ret = err
```

Comment:
Print or return error. If you want to introduce a batch operation where each loop iteration can produce error that should not stop loop you need to collect errors. Right now in production mode only last error will be displayed even if there are more.

Reply:
Currently, this debug logs every error, and returns the last one. The idea is that upper functions will return upon receiving an error, but we will have visibility of every error when in debug mode. Additionally, this pattern is consistent with other parts of the codebase. I can switch the log to an error level instead of debug so we have more visibility, but grouping all the errors into a slice wouldn't offer any additional benefits here

Reply:
Good point, will return a combined error. Slices won't offer any additional benefits given how we use/propagate Synthesizer errors, but I can get all the different errors into a single message and return once.